data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostStateValidators.java

```diff
@@ -0,0 +1,148 @@
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.AsyncApiResponse;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiEndpoint;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+
+public class PostStateValidators extends RestApiEndpoint {
+  public static final String ROUTE = "/eth/v1/beacon/states/{state_id}/validators";
+
+  private static final SerializableTypeDefinition<ObjectAndMetaData<List<StateValidatorData>>>
+      RESPONSE_TYPE =
+          SerializableTypeDefinition.<ObjectAndMetaData<List<StateValidatorData>>>object()
+              .name("GetStateValidatorsResponse")
+              .withField(
+                  EXECUTION_OPTIMISTIC, BOOLEAN_TYPE, ObjectAndMetaData::isExecutionOptimistic)
+              .withField(FINALIZED, BOOLEAN_TYPE, ObjectAndMetaData::isFinalized)
+              .withField("data", listOf(STATE_VALIDATOR_DATA_TYPE), ObjectAndMetaData::getData)
+              .build();
+
+  private static final DeserializableTypeDefinition<RequestBody> REQUEST_TYPE =
+      DeserializableTypeDefinition.object(RequestBody.class)
+          .name("PostStateValidatorsRequestBody")
+          .initializer(RequestBody::new)
+          .withField(
+              "ids",
+              DeserializableTypeDefinition.listOf(STRING_TYPE),
+              RequestBody::getIds,
+              RequestBody::setIds)
+          .withField(
+              "statuses",
+              DeserializableTypeDefinition.listOf(STRING_TYPE),
+              RequestBody::getStringStatuses,
+              RequestBody::setStatuses)
+          .build();
+
+  private final ChainDataProvider chainDataProvider;
+
+  public PostStateValidators(final DataProvider dataProvider) {
+    this(dataProvider.getChainDataProvider());
+  }
+
+  PostStateValidators(final ChainDataProvider provider) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("postStateValidators")
+            .summary("Get validators from state")
+            .description(
+                "Returns filterable list of validators with their balance, status and index.")
+            .pathParam(PARAMETER_STATE_ID)
+            .requestBodyType(REQUEST_TYPE)
+            .tags(TAG_BEACON)
+            .response(SC_OK, "Request successful", RESPONSE_TYPE)
+            .withNotFoundResponse()
+            .build());
+    this.chainDataProvider = provider;
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final RequestBody requestBody = request.getRequestBody();
+
+    final List<String> validators = requestBody.getIds();
+    final List<StatusParameter> statusParameters = requestBody.getStatuses();
```

Comment:
after the typedef changes for optional, we'll need to flow that through here too...

Reply:
we'd want test cases showing empty working too, nowhere really to comment on that currently so will put it here :)